### ISSUE TITLE
Code duplication

### ISSUE THREAD

**Issue by bnmnetp** [riknos314/RunestoneJSComponents#16]
I'm seeing that you are just copy/pasting RunestoneBase into each new component.    We don't want to do this.  You should be using good object oriented programming so that we are using one single copy of the parent classes across all of the different components.

There are some third party libraries that allow us to essentially import one js file into another, but you will have to do a bit of research on the best way to incorporate this into our set of files.

In the end, what I would like to be able to do as part of a final 'build' process is to combine all of the js files for all of the directives into one single minified runestone.js file.

Brad

**Comment by riknos314** [comment]
I separated out the RunestoneBase functions into a separate runestoneBase.js file.  As long as this file is included in the head before any js file that needs to call it everything will still work, no imports required.

This may not be ideal because it requires the js files to be loaded in a certain order, but it does eliminate the duplicate code without adding complexity from outside libraries.

**Comment by bnmnetp** [comment]
Yeah, that is why I would like you to take  a look at browserify or require.js  These are tools that will allow us to deal with these dependencies and in the end create one big js file with everything we need.

**Comment by riknos314** [comment]
After doing some research into libraries for doing this kind of thing I would prefer to use webpack over browserify or require.js.  I really like the modular nature and flexibility of webpack, as well as its developer tools.
